### ISSUE TITLE
Device enters boot loop after making camera settings changes via the web UI

### ISSUE THREAD

**Issue by teixeluis** [jomjol/AI-on-the-edge-device#3284]
### The Problem

Using the rolling version when I try to set the camera settings via the web UI, the UI itself is slightly bloated with missing text resources:

![image](https://github.com/user-attachments/assets/46998f0f-afa9-470a-9ec5-a407fc712c88)

Making  changes through it causes many of the camera related fields in the configuration file to become blank:

```
[TakeImage]
;RawImagesLocation = /log/source
;RawImagesRetention = 15
WaitBeforeTakingPicture = 5
CamGainceiling = 
CamQuality = 
CamBrightness = 2
CamContrast = 2
CamSaturation = 0
CamSharpness = 2
CamAutoSharpness = false
CamSpecialEffect = grayscale
CamWbMode = 
CamAwb = 
CamAwbGain = 
CamAec = true
CamAec2 = true
CamAeLevel = 2
CamAecValue = 
CamAgc = 
CamAgcGain = 
CamBpc = 
CamWpc = 
CamRawGma = 
CamLenc = 
CamHmirror = false
CamVflip = false
CamDcw = 
CamDenoise = 
CamZoom = true
CamZoomOffsetX = 
CamZoomOffsetY = -176
CamZoomSize = 17
LEDIntensity = 8
Demo = false
```

After that the camera becomes unstable, rebooting all the time after a few minutes of uptime. Reverting to the previous settings, it becomes stable again.

### Version

rolling (Commit: d1c815c)

### Logfile

```shell
[0d00h00m04s] 2024-09-30T20:10:08	<INF>	[MAIN] CPU frequency: 160 MHz
[0d00h00m05s] 2024-09-30T20:10:09	<INF>	[SDCARD] Folder/file presence check started...
[0d00h00m05s] 2024-09-30T20:10:09	<INF>	[SDCARD] Folder/file presence check successful
[0d00h00m05s] 2024-09-30T20:10:09	<INF>	[MAIN] Development-Branch: rolling (Commit: d1c815c), Date/Time: 2024-09-10 20:43, Web UI: Development-Branch: rolling (Commit: d1c815c)
[0d00h00m05s] 2024-09-30T20:10:09	<WRN>	[MAIN] Reset reason: Exception/panic
[0d00h00m05s] 2024-09-30T20:10:09	<WRN>	[MAIN] Device was rebooted due to a software exception! Log level is set to DEBUG until the next reboot. Flow init is delayed by 5 minutes to check the logs or do an OTA update
[0d00h00m05s] 2024-09-30T20:10:09	<WRN>	[MAIN] Keep device running until crash occurs again and check logs after device is up again
[0d00h00m05s] 2024-09-30T20:10:09	<INF>	[LOGFILE] Set log level to DEBUG
[0d00h00m05s] 2024-09-30T20:10:09	<INF>	[WLANINI] SSID: griffinnet-zh-ext
[0d00h00m05s] 2024-09-30T20:10:09	<INF>	[WLANINI] Password: XXXXXXXX
[0d00h00m05s] 2024-09-30T20:10:09	<INF>	[WLANINI] Hostname: ovalesublime-watermeter
[0d00h00m05s] 2024-09-30T20:10:09	<INF>	[WLANINI] RSSIThreshold: -75
[0d00h00m05s] 2024-09-30T20:10:09	<INF>	[MAIN] WLAN config loaded, init WIFI...
```

### Expected Behavior

Stable operation even if certain camera settings are not filled (supposedly assuming defaults). 

Web UI behaving correctly without missing resources or error messages.

### Screenshots

Also when opening the configuration page, these errors popup:

![image](https://github.com/user-attachments/assets/eb8d175f-07fa-4aa4-a8ee-1e1b29ad40b8)

**Comment by SybexX** [comment]
current rolling version: https://github.com/jomjol/AI-on-the-edge-device/actions/runs/11112677088

**Comment by teixeluis** [comment]
@SybexX I am on the latest rolling version now, and the same behavior still occurs:

![image](https://github.com/user-attachments/assets/2f95c8fb-c33e-4e13-8d58-3638f03abe33)

**Comment by SybexX** [comment]
what is in your config.ini? 
Such entries usually cause a bootloop: 
CamBpc = 
CamWpc = 
CamRawGma =

**Comment by teixeluis** [comment]
@SybexX here is the complete file:

[config_ovalesublime_watermeter_30092024.zip](https://github.com/user-attachments/files/17250655/config_ovalesublime_watermeter_30092024.zip)

And yes doing the edit with the broken web UI causes those properties to become blank.

**Comment by SybexX** [comment]
Unfortunately there was still an error in the migration, the following fix should solve the problem.
https://github.com/jomjol/AI-on-the-edge-device/actions/runs/11168328736

Please delete all parameters that have no value behind them(CamZoomOffsetX = .........) and try again.

**Comment by caco3** [comment]
@teixeluis does it work now?
Can we close the issue?

**Comment by teixeluis** [comment]
@caco3 I still didn't have the chance of testing it, but looking forward to do it this week.

**Comment by teixeluis** [comment]
Regarding the suggestion from @SybexX

> Please delete all parameters that have no value behind them(CamZoomOffsetX = .........) and try again.

that did prevented the bootloop.

Regarding the web UI errors, are these not supposed to happen with the rolling version? What steps am I missing in order to prepare the web UI package from the source?

Currently I am just creating a zip file from sdcard/html and OTA updating it on the device.

Cheers

**Comment by teixeluis** [comment]
> The HTML files which are contained in the repo are used as templates. During the build process, they get extended/filled.
> SO, if you want to build it yourself, I suggest to fork the repo, add the modifications and let it build using the Github actions.
>

@caco3 what about if I don't have any modifications of my own to the html files, where can I get the built html package associated to the rolling branch, or what steps can I execute to generate it locally? Do I need to fork it and bulld in github actions anyway, or is there a place where I can download it prebuilt?

**Comment by caco3** [comment]
The Github actions build the package for every commit we push to github.
Just go to https://github.com/jomjol/AI-on-the-edge-device/actions and select the build you want.
Either scroll through the list (newest on top) or select the branch the top right:
![image](https://github.com/user-attachments/assets/9f004226-c2f9-4315-aceb-0e88ebed664c)
Note that since yesterday we renamed `rolling` to `main`.

Once you have selected the build, scoll down and pick the right zip file.
We do not provide a separate package just for the UI. But any zip file contains it.

**Comment by teixeluis** [comment]
tx

**Comment by SybexX** [comment]
You only need to update the HTML files if they have been changed, otherwise it is sufficient if you just update the firmware.bin.
@caco3 can't he use AI-on-the-edge-device-main/tools/parameter-tooltip-generator/generate-param-doc-tooltips.py for local building?

**Comment by caco3** [comment]
> @caco3 can't he use AI-on-the-edge-device-main/tools/parameter-tooltip-generator/generate-param-doc-tooltips.py for local building?

Yes he can, but he needs to setup python and the needed dependencies.
Letting Github do it usually is just easier.
Also note that this only generates the tooltps, but there are more modifications happening on the HTML files 8mainly setting the version/commit hash).

As said by @SybexX before, you can use the UI of another version. The changes are usually minor or even zero. But you will see a warning.

**Comment by teixeluis** [comment]
All looking good with the build from the main branch. OTA updated with the full zip file (AI-on-the-edge-device__update__main_(57db60f).zip) and UI is ok and managed to setup the recogniton again. Using dig-cont_0710_s3_q.tflite and recognition is correct so far.
